src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionDiagnosticAnalyzer.cs

```diff
@@ -75,6 +75,7 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
 
             var additionalLocations = ArrayBuilder<Location>.GetInstance();
             additionalLocations.Add(switchStatement.GetLocation());
+            additionalLocations.AddOptional(variableSymbolOpt?.Locations.FirstOrDefault());
```

Comment:
note: it's much safer to use `variableSymbolOpt?.DeclaratingSyntaxReferences` the syntax references are guaranteed to round-trip to the symbol using GetDeclaredSymbol, the `Locations` are not.